docs/data-model.md

```diff
@@ -60,6 +62,27 @@ erDiagram
     string trn FK "Primary key for people in DQT"
   }
 
+  User {
+    uuid id PK
+    string email UK
+  }
+
+  Membership {
+    uuid id PK
+    uuid user_id FK
+    string organisation_type FK "Polymorphic association with School or Provider"
+    string organisation_id FK "Polymorphic association with School or Provider"
+    enum service "Placements or Claims"
```

Comment:
After our discussion, I think it's best if we move this enum into the `User` table.

Once a user signs in, we need to know if the user is logged in from the Claims DSI or Placements DSI.

With a split User on both sides, we can easily find our user count.

Reply:
I might not have all the context here but having the 'service enum' on the membership will allow 1 user access to both services through multiple memberships. If we need to know from which service the user logs in we can use some other column? Just a thought

Reply:
We've considered the likelihood that a user may have no memberships when they first log in.

Reply:
I think either approach could be made to work. I've been on the fence about whether 'service' should live on the User or the Membership. But having given it some discussion and thought, I think User is probably the right place for it.

This would support User models in each service which look something like this (pseudo code):

```ruby
class User {
  validates :email, uniqueness: { scope: :service }
}

class Claims::User < User {
  default_scope { where service: "claims" }
}

class Placements::User < User {
  default_scope { where service: "placements" }
}
```

Then we'd easily be able to do things like this, without having to think about the service:

```ruby
Claims::User.where(email: "john.smith@example.com").exists?
Claims::User.count
# etc.
```

Since we're also going to need different DfE Sign-in client IDs, it seems reasonable to carry this separation through to the User model.

Reply:
For what it's worth, I think the same approach could be carried forward for Schools too.

```ruby
class School {
  validates :urn, presence: true
}

class Claims::School {
  default_scope { where claims_enabled: true }
}

class Placement::School {
  default_scope { where placements_enabled: true }
}
```

And then when we need to list all schools within our services, we can simply use:

```ruby
Claims::School.all
```

Reply:
@Nitemaeric are you happy with this? (Will squash before merge)

https://github.com/DFE-Digital/itt-mentor-services/pull/30/commits/21b2fbf25e89eb1cc6a4646d90c28411e49ca37a

Reply:
Yeah, I like the idea of domain scoped models either pointing to the shared model or applying a scope on the shared model.